### PR TITLE
Adding New Experience previewing Overlay

### PR DESCRIPTION
**Pull request by JustinLohKingWei**
### Summary 
This PR implements the ability to preview actions through a toggle-able preview overlay.
### Details
1. Adds a new preview overlay to the navbar 
2. Overlay is toggle-able with the Preview Overlay
3. Preview Overlay allows users to traverse different actions from a saved sequence of actions (save button MUST be pressed)
### How to test?
1. spin up the app
2. select an apparatus 
3. Select a sequence of actions and save in the overlay
4. Select "Toggle Preview" to bring up the preview overlay
5. use the left and right arrows to see if the sequence matches the saved ones
### Checks 
- [ ] Tested Changes
- [ ] Stakeholder Approval

### CODE REVIEW

**Comment by mckenzietsean** [comment]
Works great. Aside from the 4 code smells to fix, my only major complaint would be that preview starts at sequence 0, but it doesn't start its animation until you select sequence 0 again.

File: src/components/previewOverlay/PreviewOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        <PreviewGridLeft>
+          <IconButton
+            style={{
+              minWidth: "100%",
+              minHeight: "100%",
+              outlineStyle: "solid",
+              outlineWidth: "0.1em",
+              outlineColor: "black",
+            }}
+            onClick={selected > 0 ? ()=>{setCount(selected - 1); callToWebGL(actionList[selected-1][0], actionList[selected-1][1]) } :undefined}
```
Comment (Ysabz) [inline src/components/previewOverlay/PreviewOverlay.tsx:94]: nested ternary operation is a  code smell and causes confusion. It's better to simplify this into an independent statement.

File: src/components/previewOverlay/PreviewOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        <PreviewGridRight>
+          <IconButton
+            style={{
+              minWidth: "100%",
+              minHeight: "100%",
+              outlineStyle: "solid",
+              outlineWidth: "0.1em",
+              outlineColor: "black",
+            }}
+            onClick={ selected < actionList.length-1 ? ()=>{setCount(selected + 1) ; callToWebGL(actionList[selected+1][0], actionList[selected+1][1]) } :undefined}
```
Comment (Ysabz) [inline src/components/previewOverlay/PreviewOverlay.tsx:119]: same issue with the ternary operation here

File: src/components/previewOverlay/PreviewOverlay.tsx (hunk 1 above)
Comment (JustinLohKingWei) [reply src/components/previewOverlay/PreviewOverlay.tsx:94]: There are a few similar code smells in Overlay.tsx as well which i did not write. Should i correct those as well?

**Comment by JustinLohKingWei** [comment]
> Works great. Aside from the 4 code smells to fix, my only major complaint would be that preview starts at sequence 0, but it doesn't start its animation until you select sequence 0 again.

Unfortunately. This may have to be classified as a seperate bug to fix

**Comment by JustinLohKingWei** [comment]
Code smells have been eliminated @Ysabz @mckenzietsean

File: src/components/previewOverlay/PreviewOverlay.tsx (hunk 1 above)
Comment (JustinLohKingWei) [reply src/components/previewOverlay/PreviewOverlay.tsx]: all code smells are resolved

**Review by mckenzietsean — APPROVED** [review]
Aside for the previously discussed bug, everything else is perfect.

**Pull request merged**